### PR TITLE
featureimage/IPMVAL-3756-endu-graphs

### PR DESCRIPTION
**Pull request by FrancoisRegisDegott-eaton**
* rework some files, add logs and lap times...
* fix remaining_ttl computation on metric rewrite in shm
* fix MB timeout root cause: remove blocking 'flush to DB' from server actor handling the MB (flush is done by metrics_pull actor)
* fix sql request root cause (select measurements on topic_id (int) instead of topic (string))

### CODE REVIEW

File: src/fty_metric_store_server.cc [hunk 1]
```diff
@@ -262,6 +286,9 @@ static zmsg_t* s_process_mailbox_aggregate(mlm_client_t* /*client*/, zmsg_t** me
     zstr_free(&cmd);
     zmsg_destroy(message_p);
 
+    lap = zclock_mono() - lap;
+    log_info("OUT handle MAILBOX aggregate, lap: %ju ms", lap);
```
Comment (ericclappier-eaton) [inline src/fty_metric_store_server.cc]: Maybe debug or trace level

File: src/fty_metric_store_server.cc [hunk 2]
```diff
@@ -191,23 +206,27 @@ static zmsg_t* s_process_mailbox_aggregate(mlm_client_t* /*client*/, zmsg_t** me
...
-        log_debug("select topic (rv: %d, topic: '%s', units: '%s')", rv, topic.c_str(), units.c_str());
+        log_trace("select topic, rv: %d, exists: %s, topic: '%s', id: %d, units: '%s'",
+            rv, (topic_exists ? "true" : "false"), topic.c_str(), topic_id, topic_units.c_str());
 
         if (rv != 0) {
-            if (rv == -2) {
-                log_error("topic is not found (%s)", topic.c_str());
-                SET_ERROR_MSG_AND_BREAK("BAD_REQUEST");
-            }
+            //if (rv == -2) {
```
Comment (ericclappier-eaton) [inline src/fty_metric_store_server.cc:222]: May be add if else block to manage BAD_REQUEST

File: src/fty_metric_store_server.cc [hunk 3]
```diff
@@ -112,6 +113,16 @@ static zmsg_t* s_process_mailbox_aggregate(mlm_client_t* /*client*/, zmsg_t** me
     char* end_date_str   = zmsg_popstr(msg);
     char* ordered        = zmsg_popstr(msg);
 
+    log_info("IN handle MAILBOX aggregate: asset: %s, quantity: %s, step: %s, aggr_type: %s, start: %s, end: %s, ordered: %s",
```
Comment (ericclappier-eaton) [inline src/fty_metric_store_server.cc]: May be debug level trace

File: src/fty_metric_store_server.cc [hunk 4]
```diff
@@ -451,35 +494,51 @@ static void s_process_pull_store_shm_metrics(fty::shm::shmMetrics& metrics)
...
+            scale,
+            int64_t(_time),
+            fty_proto_unit(m),
+            fty_proto_name(m)
+        );
+
+        // inserted, flag this metric (write in shm)
+        auto now = uint64_t(time(nullptr));
+        if ((fty_proto_time(m) + fty_proto_ttl(m)) < now) {
+            uint32_t remaining_ttl = uint32_t(fty_proto_time(m) + fty_proto_ttl(m) - now);
```
Comment (ericclappier-eaton) [inline src/fty_metric_store_server.cc]: May be static_cast better for all cast

File: src/fty_metric_store_server.cc [hunk 5]
```diff
@@ -489,15 +548,20 @@ void fty_metric_store_metric_pull(zsock_t* pipe, void* /*args*/)
             }
 
             if (zpoller_expired(poller)) {
-                log_debug("read metrics from shm");
+                log_trace("IN metrics pull");
+                int64_t lap = zclock_mono();
+
                 fty::shm::shmMetrics result;
                 fty::shm::read_metrics(".*", ".*", result);
-                log_debug("metric reads : %d", result.size());
```
Comment (ericclappier-eaton) [inline src/fty_metric_store_server.cc:495]: May be add the size in the following trace

File: src/fty_metric_store_server.cc [hunk 6]
```diff
@@ -451,35 +494,51 @@ static void s_process_pull_store_shm_metrics(fty::shm::shmMetrics& metrics)
...
+            if( cntDbConnectFailed++ == 0) {
+                log_error("Can't connect to the database");
+            }
             continue;
         }
 
-        // time is a time when message was received
+        // time is the time when message was received
         uint64_t _time = fty_proto_time(m);
         insert_into_measurement(
```
Comment (ericclappier-eaton) [inline src/fty_metric_store_server.cc]: No read error

**Review by ericclappier-eaton — DISMISSED** [review]

**Review by ericclappier-eaton — DISMISSED** [review]

File: src/fty_metric_store_server.cc (hunk 6 above)
Comment (FrancoisRegisDegott-eaton) [reply src/fty_metric_store_server.cc]: done

File: src/fty_metric_store_server.cc (hunk 3 above)
Comment (FrancoisRegisDegott-eaton) [reply src/fty_metric_store_server.cc]: done

File: src/fty_metric_store_server.cc (hunk 1 above)
Comment (FrancoisRegisDegott-eaton) [reply src/fty_metric_store_server.cc]: done

File: src/fty_metric_store_server.cc (hunk 5 above)
Comment (FrancoisRegisDegott-eaton) [reply src/fty_metric_store_server.cc:495]: size is logged in s_process_pull_store_shm_metrics()

File: src/fty_metric_store_server.cc (hunk 4 above)
Comment (FrancoisRegisDegott-eaton) [reply src/fty_metric_store_server.cc]: kept as it was... (that ttl value is positive and low related to max uint32)

**Review by perrettecl — DISMISSED** [review]
We should plan an update of "logs" macro here.

**Review by perrettecl — APPROVED** [review]

**Pull request merged**